src/link.js

```diff
@@ -1,37 +1,117 @@
 import React, { Component, PropTypes } from 'react';
+
 import { PUSH, REPLACE } from './action-types';
 
-export default class Link extends Component {
-  onClick(event) {
-    event.preventDefault();
-    const { replaceState, dispatch } = this.props;
+const LEFT_MOUSE_BUTTON = 0;
+
+const normalizeLocation = href => {
+  if (typeof href === 'string') {
+    const [pathname, query] = href.split('?');
+    return query ? { pathname, search: `?${query}` } : { pathname };
+  }
+  return href;
+};
+
+const resolveQueryForLocation = ({ linkLocation, persistQuery, currentLocation }) => {
+  const currentQuery = currentLocation &&
+    currentLocation.query;
+
+  // Only use the query from state if it exists
+  // and the href doesn't provide its own query
+  if (
+    persistQuery &&
+    currentQuery &&
+    !linkLocation.search &&
+    !linkLocation.query
+  ) {
+    return { pathname: linkLocation.pathname, query: currentQuery};
+  }
+
+  return linkLocation;
+};
+
+const clickedWithModifier = e =>
+  e.button === LEFT_MOUSE_BUTTON &&
```

Comment:
I don't *think* you want to check for `LEFT_MOUSE_BUTTON` here. (I haven't tested this, so could be wrong).

This behavior should happen if you're clicking a link by pressing enter on the keyboard as well.

We'll also want to check for right clicks and not trigger any behavior on them. This module includes some logic that handles this. Don't know if we want to use the whole thing or not because it does some other stuff too: https://github.com/lukekarrys/local-links/blob/master/local-links.js#L26-L28